compiler/tests/solidity.rs

```diff
@@ -0,0 +1,44 @@
+//! Solidity tests that help us prove assumptions about how Solidty handles
+//! certain things
+
+#![cfg(feature = "solc-backend")]
+use rstest::rstest;
+
+mod utils;
+use utils::*;
+
+#[rstest(
+    method,
+    reason,
+    case("revert_me", "Not enough Ether provided."),
+    case(
+        "revert_with_long_string",
+        "A muuuuuch longer reason string that consumes multiple words"
+    ),
+    case("revert_with_empty_string", "")
+)]
+fn test_revert_string_reason(method: &str, reason: &str) {
+    with_executor(&|mut executor| {
+        let harness = deploy_solidity_contract(&mut executor, "revert_test.sol", "Foo", &[]);
+
+        let exit = harness.capture_call(&mut executor, method, &[]);
+
+        let expected_reason = format!("0x{}", hex::encode(encode_error_reason(reason)));
```

Comment:
This is proving that our `encode_error_reason` is producing the same encoding that solidity does